app/components/SearchMapListItem.js

```diff
@@ -11,19 +11,19 @@ class SearchMapListItem extends React.Component {
   }
 
   selectSchool() {
-    SearchActionCreators.selectSchool(this.props.school);
+    SearchActionCreators.selectSchool(this.props.school.location.id);
```

Comment:
You're right. The same goes for the variable names of the search results especially in map views where they are referred to as schools which might be confusing, because the data they contain differs slight from other school data.